app/Helpers/Get/LoadDataHelper.php

```diff
@@ -329,4 +296,33 @@ public static function getCities ($country, string $state) : array
             'cities' => $cities
         ];
     }
+
+    /**
+     * Gets the users from the given ids
+     * Attaches to each their global or location-based XP
+     * Formats them for display on the leaderboards
+     *
+     * @param $leaderboardIds
+     * @return array
+     */
+    protected static function getLeadersFromLeaderboards($leaderboardIds): array
+    {
+        $users = User::query()
+            ->whereIn('id', array_keys($leaderboardIds))
```

Comment:
If we do that we'll have to take into consideration what we do in the `getLeaders()` function, which is not ideal. We already filter columns for output, so that is solved. Also fetching data for 10 users is not a problem.